python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -1270,6 +1270,10 @@ def set_download_parameters(self, params, relative_path, target_file=None, check
         params["minrate"] = self.minrate
         params['proxies'] = get_proxies(self.proxy_url, self.proxy_user, self.proxy_pass)
 
+        if len(self.repo.urls) == 1:
+            # when we have only 1 url copy it to have a 2nd try
+            params['urls'].append(self.repo.urls[0])
```

Comment:
Wouldn't it be nicer to set the "retry" option to 2 instead of duplicating the URL?

Reply:
I think we're talking about different "retry" options then. [urlgrabber's default is `None`](https://github.com/rpm-software-management/urlgrabber/blob/master/urlgrabber/grabber.py#L1055), which disabled re-trying.

Reply:
I think we would need to change also the "retrycodes" as it would not retry on HTTP errors.

Reply:
please check the new commit if this is what you had in mind